test/supervisor_test.exs

```diff
@@ -0,0 +1,43 @@
+defmodule KafkaEx.Supervisor.Test do
+  use ExUnit.Case, async: false
+
+  test "uses configured max_restarts" do
+    # stop the application, configure supervision max_restarts and restart the application
+    configured_max_restarts = 15
+
+    :ok = Application.stop(:kafka_ex)
+    Application.put_env(:kafka_ex, :max_restarts, configured_max_restarts)
+    {:ok, _} = Application.ensure_all_started(:kafka_ex)
+    [max_restarts, _max_seconds] = :sys.get_state(KafkaEx.Supervisor) |> elem(9)
```

Comment:
Using `elem` to dig into the supervisor's state is prone to breaking if the structure changes.  Being just test code it's not TOO bad, but maybe better embedded in a function?  I looked at the supervisor source code and don't see any way to introspect the state and had no luck :/

TBH I'd be comfortable not testing this.  I'm a big believer in testing, but this seems like a lot of work to test something pretty straightforward.